tests/e2e.test.ts

```diff
@@ -0,0 +1,242 @@
+import { describe, it, expect, beforeAll, afterAll, afterEach } from 'vitest'
+
+import fs from 'fs'
+import path from 'path'
+import { ExecException, exec } from 'child_process'
+
+/**
+ * NOTE: This is the compiled CLI for this package to test that everything works end-to-end
+ * and avoid bugs like this: https://github.com/TimMikeladze/tsc-baseline/issues/21
+ *
+ * You'll need to compile the CLI with `yarn dev` and make sure it successfully compiles the
+ * latest changes to be able to test them here.
+ */
+const CLI_PATH = path.resolve(__dirname, '../dist/cli.module.js')
+
+let tempDir: string
+
+const getBaselinePath = () => path.resolve(tempDir, '.tsc-baseline.json')
+
+type CliOutput = {
+  code: number
+  error: ExecException | null
+  stderr: string
+  stdout: string
+}
+
+/**
+ * A utility function for running the compiled CLI.
+ *
+ * @param input The CLI input command and arguments
+ * @param stdin The string input passed to stdin for the CLI
+ * @param cwd The directory to run the command from
+ * @returns The output of the CLI including exit code, output error, stdout, and stderr
+ *
+ * Tips for using this:
+ * - The input piped to stdin from the TypeScript output is not the same as what is
+ *   seen in the terminal. To get this value on mac use `yarn type-check | pbcopy`
+ * - Make sure the CLI has properly compiled and is up-to-date before running tests
+ *   in this file.
+ * - Logging in the CLI won't show up in the test, it will only show up in the
+ *   returned `stdout`/`stderr` from this function
+ */
+const cli = (
+  input: string,
+  stdin: string | null = null,
+  cwd = tempDir
+): Promise<CliOutput> =>
+  new Promise((resolve) => {
+    const cliProcess = exec(
+      `node ${CLI_PATH} ${input}`,
+      { cwd },
+      (error, stdout, stderr) => {
+        resolve({
+          code: error && error.code ? error.code : 0,
+          error,
+          stdout,
+          stderr
+        })
+      }
+    )
+    // NOTE: Currently running this cli utility without stdin isn't fully supported
+    // because process.stdin.isTTY (which is currently used to check if the CLI is
+    // run without stdio) isn't defined for child processes: https://github.com/nodejs/node/issues/51750
```

Comment:
I was struggling to figure out how to get this working properly. Despite this limitation, I think it might still be valuable to merge this as-is for now and then come back and update it if we want to test the `clear` command in the future.